ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js

```diff
@@ -1,33 +1,129 @@
         leftIcon: 'fas fa-eye',
-        label: t('eth_accounts'),
+        label: t('permission_ethereumAccounts'),
+        rightIcon: null,
+      },
+      ///: BEGIN:ONLY_INCLUDE_IN(flask)
+      [RestrictedMethods.snap_confirm]: {
+        leftIcon: 'fas fa-user-check',
+        label: t('permission_customConfirmation'),
+        rightIcon: null,
+      },
+      [RestrictedMethods['snap_getBip44Entropy_*']]: (permissionName) => {
+        const coinType = permissionName.split('_').slice(-1);
+        return {
+          leftIcon: 'fas fa-door-open',
+          label: t('permission_manageBip44Keys', [
+            coinTypeToProtocolName(coinType) ||
+              `${coinType} (Unrecognized protocol)`,
+          ]),
+          rightIcon: null,
+        };
+      },
+      [RestrictedMethods.snap_manageState]: {
+        leftIcon: 'fas fa-download',
+        label: t('permission_manageState'),
+        rightIcon: null,
+      },
+      [RestrictedMethods['wallet_snap_*']]: (permissionName) => {
+        const snapId = permissionName.split('_').slice(-1);
+        return {
+          leftIcon: 'fas fa-bolt',
+          label: t('permission_accessSnap', [snapId]),
+          rightIcon: null,
+        };
+      },
+      [EndowmentPermissions['endowment:network-access']]: {
+        leftIcon: 'fas fa-wifi',
+        label: t('permission_accessNetwork'),
         rightIcon: null,
       },
+      ///: END:ONLY_INCLUDE_IN
+      [UNKNOWN_PERMISSION]: (permissionName) => {
+        return {
+          leftIcon: 'fas fa-times-circle',
+          label: t('permission_unknown', [permissionName ?? 'undefined']),
+          rightIcon: null,
+        };
+      },
     };
   }, [t]);
 
   return (
     <div className="permissions-connect-permission-list">
-      {Object.keys(permissions).map((permission) => (
-        <div className="permission" key={PERMISSION_TYPES[permission].label}>
-          <i className={PERMISSION_TYPES[permission].leftIcon} />
-          {PERMISSION_TYPES[permission].label}
-          <i className={PERMISSION_TYPES[permission].rightIcon} />
-        </div>
-      ))}
+      {Object.keys(permissions).map((permission) => {
+        const listValue =
+          PERMISSION_LIST_VALUES[
+            getPermissionKey(permission, PERMISSION_LIST_VALUES)
+          ];
+
+        const { label, leftIcon, rightIcon } =
+          typeof listValue === 'function' ? listValue(permission) : listValue;
+
+        return (
+          <div className="permission" key={permission}>
+            <i className={leftIcon} />
+            {label}
+            {rightIcon && <i className={rightIcon} />}
+          </div>
+        );
+      })}
     </div>
   );
 }
 
 PermissionsConnectPermissionList.propTypes = {
-  permissions: PropTypes.objectOf(PropTypes.bool).isRequired,
+  permissions: PropTypes.object.isRequired, // TODO(ritave): Currently some properties are boolean, some are {} empty objects, fix the issue at root cause.
```

Comment:
I think they are supposed to be objects in all cases. At least according to the skunkworks types.

Reply:
Correct, but `getRequestedMethodState` in `permission-page-container.component` will return an object where all values are `true`. If we think that this is a problem we should fix the aforementioned method on `develop`. For now, I will remove this TODO.